common/hardware_tici.py

```diff
@@ -1,62 +1,134 @@
-import serial
-
 from common.hardware_base import HardwareBase
 from cereal import log
 import subprocess
 
+NM = 'org.freedesktop.NetworkManager'
+NM_CON_ACT = NM + '.Connection.Active'
+NM_DEV_WL = NM + '.Device.Wireless'
+NM_AP = NM + '.AccessPoint'
+DBUS_PROPS = 'org.freedesktop.DBus.Properties'
+
+MM = 'org.freedesktop.ModemManager1'
+MM_MODEM = MM + ".Modem"
+MM_MODEM_SIMPLE = MM + ".Modem.Simple"
+MM_SIM = MM + ".Sim"
+
+MM_MODEM_STATE_CONNECTED = 11
 
 NetworkType = log.ThermalData.NetworkType
 NetworkStrength = log.ThermalData.NetworkStrength
 
-
-def run_at_command(cmd, timeout=0.1):
-  with serial.Serial("/dev/ttyUSB2", timeout=timeout) as ser:
-    ser.write(cmd + b"\r\n")
-    ser.readline()  # Modem echos request
-    return ser.readline().decode().rstrip()
+# https://developer.gnome.org/ModemManager/unstable/ModemManager-Flags-and-Enumerations.html#MMModemAccessTechnology
+MM_MODEM_ACCESS_TECHNOLOGY_EDGE = 1 << 4
+MM_MODEM_ACCESS_TECHNOLOGY_LTE = 1 << 14
 
 
 class Tici(HardwareBase):
-  def get_sound_card_online(self):
-    return True
+  def __init__(self):
+    import dbus  # pylint: disable=import-error
 
-  def get_imei(self, slot):
-    if slot != 0:
-      return ""
-
-    for _ in range(10):
-      try:
-        imei = run_at_command(b"AT+CGSN")
-        if len(imei) == 15:
-          return imei
-      except serial.SerialException:
-        pass
+    self.bus = dbus.SystemBus()
+    self.nm = self.bus.get_object(NM, '/org/freedesktop/NetworkManager')
+    self.mm = self.bus.get_object(MM, '/org/freedesktop/ModemManager1')
 
-    raise RuntimeError("Error getting IMEI")
+  def get_sound_card_online(self):
+    return True
 
   def get_serial(self):
     return self.get_cmdline()['androidboot.serialno']
 
-  def get_subscriber_info(self):
-    return ""
-
   def reboot(self, reason=None):
     subprocess.check_output(["sudo", "reboot"])
 
   def get_network_type(self):
-    return NetworkType.wifi
+    # conns = self.nm.Get(NM, 'ActiveConnections', dbus_interface=DBUS_PROPS)
+    primary_connection = self.nm.Get(NM, 'PrimaryConnection', dbus_interface=DBUS_PROPS)
+    primary_connection = self.bus.get_object(NM, primary_connection)
+    tp = primary_connection.Get(NM_CON_ACT, 'Type', dbus_interface=DBUS_PROPS)
+
+    if tp in ['802-3-ethernet', '802-11-wireless']:
+      return NetworkType.wifi
+    elif tp in ['gsm']:
+      modem = self.get_modem()
+      access_t = modem.Get(MM_MODEM, 'AccessTechnologies', dbus_interface=DBUS_PROPS)
+      if access_t >= MM_MODEM_ACCESS_TECHNOLOGY_LTE:
```

Comment:
I am testing with LTE SIM (verified) and it registers with 3G only (UMTS), I see this recent commit and I belive it does not detect modem properly as LTE modem, so that it is effectively 3G only